### PR TITLE
ARROW-3037: [Go] implement Null array

### PR DESCRIPTION
**Pull request by sbinet**
needs https://github.com/apache/arrow/pull/2419.

### CODE REVIEW

**Comment by sbinet** [comment]
@stuartcarnie PTAL.

**Comment by sbinet** [comment]
rebased on top of latest master.
PTAL.

**Comment by pitrou** [comment]
There are conflicts here now.

**Comment by wesm** [comment]
GitHub shows false conflicts with stacked patches in my experience

**Comment by sbinet** [comment]
rebased on top of latest master and fixed (false) conflicts.

**Review by wesm — APPROVED** [review]
+1

File: go/README.md [hunk 1]
```diff
@@ -133,7 +133,7 @@ them internally in the [ifql][] execution engine and storage layers of [InfluxDB
 - [x] Variable-length binary
 - [ ] String (valid UTF-8)
 - [ ] Half-float (16-bit)
-- [ ] Null (no physical storage)
+- [x] Null (no physical storage)
```
[the remark this answers is not in the document]
Comment (sbinet) [reply go/README.md]: Yes. Will do.

**Pull request closed without merging**